core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -2229,15 +2234,14 @@ class ReplicaManager(val config: KafkaConfig,
   }
 
   def deleteStrayReplicas(topicPartitions: Iterable[TopicPartition]): Unit = {
-    stopPartitions(topicPartitions.map { tp => tp -> true }.toMap).foreach {
-      case (topicPartition, e) =>
-        if (e.isInstanceOf[KafkaStorageException]) {
-          stateChangeLogger.error(s"Unable to delete stray replica $topicPartition because " +
-            "the local replica for the partition is in an offline log directory")
-        } else {
-          stateChangeLogger.error(s"Unable to delete stray replica $topicPartition because " +
-            s"we got an unexpected ${e.getClass.getName} exception: ${e.getMessage}", e)
-        }
+    stopPartitions(topicPartitions.map(tp => tp -> true).toMap).forKeyValue { (topicPartition, e) =>
+      if (e.isInstanceOf[KafkaStorageException]) {
+        stateChangeLogger.error(s"Unable to delete stray replica $topicPartition because " +
+          "the local replica for the partition is in an offline log directory")
```

Comment:
nit: I was kind of wondering if it's worthwhile logging the exception message in here. We do construct the exception with different messages in different scenarios.